packages/did/src/DidDetails/LightDidDetails.utils.ts

```diff
@@ -6,24 +6,67 @@
  */
 
 import { encode as cborEncode, decode as cborDecode } from 'cbor'
-import type { LightDidDetailsCreationOpts } from './LightDidDetails'
+import { SDKErrors } from '@kiltprotocol/utils'
+import type { LightDidDetailsCreationOpts } from '../types'
+import { getEncodingForSigningKeyType, parseDidUrl } from '../Did.utils'
 
 const ENCRYPTION_KEY_MAP_KEY = 'e'
+const SERVICES_KEY_MAP_KEY = 's'
+
+export function checkLightDidCreationOptions(
+  options: LightDidDetailsCreationOpts
+): void {
+  // Check authentication key type
+  const authenticationKeyTypeEncoding = getEncodingForSigningKeyType(
+    options.authenticationKey.type
+  )
+  if (!authenticationKeyTypeEncoding) {
+    throw SDKErrors.ERROR_UNSUPPORTED_KEY
+  }
+
+  // Check service endpoints
+  if (!options.serviceEndpoints) {
+    return
+  }
+
+  options.serviceEndpoints?.forEach((service) => {
+    try {
+      parseDidUrl(service.id)
+      throw new Error(
+        `Invalid service ID provided: ${service.id}. The service ID should be a simple identifier and not a complete DID URI.`
+      )
+      // A service ID cannot have a reserved ID that is used for key IDs.
+      if (service.id === 'authentication' || service.id === 'encryption') {
+        throw new Error(
+          `Cannot specify a service ID with the name ${service.id} as it is a reserved keyword.`
+        )
+      }
+      // eslint-disable-next-line no-empty
+    } catch {}
```

Comment:
Q: You are throwing errors in the try block, but ignore them later? What is this for?

Reply:
Cause I was looking for a way to make sure that the input ID was not a DID URI, but a simple string identifier. So if `parseDidUrl` throws, it means that the provided string is not a valid DID URI and we are good. If it does not throw, it means that a DID URI was provided, and that is invalid as for a light DID we only expect a service ID and we build the full DID URI internally. Do you think there's a better way to achieve this?

Reply:
I added a comment -> https://github.com/KILTprotocol/sdk-js/pull/428/commits/f37420041b924b33145f930bb2491461ec5cc3dc

But if there's a better way to do it, I'm open for discussion.

Reply:
Good enough -> https://github.com/KILTprotocol/sdk-js/pull/428/commits/1954f59e995dd56d19c85d3b8c077045ac18655f